### DISCUSSION TITLE
Installing Phalcon causing kind of a timeout

### DISCUSSION THREAD

**Discussion by Ivalys** [phalcon/cphalcon#16331] [Q&A] [answered]
Hello,

I'm trying to install Phalcon on a VPS but It crashes everytime (but seems like not always with the same last message - maybe a configuration is necessary on my VPS ?), I wonder why.

Here's my setup :
- Ubuntu 22.04
- 2 GB RAM / 1 vCore
- PHP 8.1
- Got all dependencies as mentioned in Phalcon's docs (Wonder why phpize isn't listed ? I've installed php-dev manually)

The problem occurs when I'm at the PECL step. Precisely when I try this command (as root) :

`pecl install phalcon-5.0.0`

Also happens with phalcon-5.2.1.

Everything seems to work but then I'm getting a huge amount of what looks like Zend error from /tmp/. Got a log, attaching it to the subject. What am I doing wrong ?

[log.txt](https://github.com/phalcon/cphalcon/files/11355190/log.txt)

**Comment by s-ohnishi** [comment]
Hi.

I will disclose the Dockerfile that I used when I created the Docker container.
(you may not need some and it may not be the best description)
```
FROM php:8.1-fpm-buster

ENV TZ=Asia/Tokyo \
  PHP_INI_SCAN_DIR=:/etc/php.extra.d \
  DEBIAN_FRONTEND=noninteractive

RUN set -eux; \
  apt-get update; \
  mkdir -p -m 777 /etc/php.extra.d; \
  pecl install xdebug && docker-php-ext-enable xdebug; \
  pecl install psr && docker-php-ext-enable psr; \
  apt-get install -y \
    libfreetype6-dev \
    libjpeg62-turbo-dev \
    libzip-dev; \
  docker-php-ext-configure gd --with-freetype --with-jpeg; \
  docker-php-ext-install -j$(nproc) gd mysqli pdo_mysql opcache gettext zip


ENV PHALCON_VERSION="5.1.3"

RUN set -eux; \
  pecl install phalcon-${PHALCON_VERSION};  \
  pecl clear-cache

RUN set -eux; \
  mkdir /root/composer; cd /root/composer; \
  php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');"; \
  php composer-setup.php --install-dir=/usr/local/bin --filename=composer; \
  chmod +x /usr/local/bin/composer; \
  cd /root; rm -rf composer; \
  cd /

RUN set -eux; \
  pecl install mailparse && docker-php-ext-enable mailparse
```
As you can see in this, installing Phalcon is no different than what you describe.
It took a few minutes on a good PC.
I remember having to wait a long time when I ran it on a VPS with 1 GB RAM / 2 vCores.
It probably took about 7 minutes, maybe closer to 10 minutes.

**Reply by Ivalys** [reply]
Great to read ! It lasted for more than 1 hour for me. CPU/RAM was at 100%. Maybe because Apache, Fail2Ban and other services were running.

Sadly it last with that message :

`cc: fatal error: Killed signal terminated program cc1
compilation terminated.
make: *** [Makefile:202: phalcon.lo] Error 1
ERROR: `make' failed`

Don't think that's a great news tho. Any idea why it occured ?
Is it some kind of command timeout ? OOM Killer maybe ? Dunno how I should fix it atm.

**Reply by s-ohnishi** [reply]
I don't know the details, but I have also encountered compilation errors in the past.
At that time, I feel like I was missing something, but I don't remember it clearly. sorry.
Since there was a work log at the time of 5.0.0RC4, I will post the Phalcon installation part.
Debian didn't have PHP8.1 at the time, so I installed it from packages.sury.org.
I'm glad if you can use it as a reference.
```
# apt -y install apt-transport-https lsb-release ca-certificates curl wget
# apt -y install apt-transport-https
# wget -O /etc/apt/trusted.gpg.d/php.gpg https://packages.sury.org/php/apt.gpg
# echo "deb https://packages.sury.org/php/ $(lsb_release -sc) main" > /etc/apt/sources.list.d/php.list

# apt install php-dev php8.1 php8.1-fpm \
  php8.1-curl php8.1-gd php-json php8.1-mbstring php8.1-mysql php8.1-sqlite3 php8.1-opcache \
  php-psr php-pear php-xdebug php-mailparse

# pecl search phalcon
# pecl install phalcon-5.0.0RC4

# echo "extension=phalcon.so" > /etc/php/8.1/mods-available/phalcon.ini
# phpenmod -v 8.1 -s ALL phalcon
```

**Reply by Ivalys** [reply, marked as answer]
Hello,

Sadly it doesn't change anything.
I solved the issue using WSL to compile Phalcon as phalcon.so and moved it to my VPS. Solved all the problem.

Also, didn't get any message error using WSL with the same configuration as my VPS.
Strange but do remind you can compile Phalcon locally then put it on your server ! :)

**Reply by s-ohnishi** [reply]
I'm sorry I can't help you.
I also tried it on VPS, but I got the same error.
It takes about 30 minutes to get an error, so it looks like it's timed out.
(there is no "timeout" text on the screen)

When I tried to install PHP8.1 on Debian 11.7 and use sury.org, PHP8.2 was installed and it took me a long time to fix it.
I was able to revert to PHP8.1, but the error has not been resolved yet.

With Docker, I was able to install it in the official PHP8.1 container, so I'm guessing that the necessary packages are missing.
If I can't solve it by any means, I would like to adopt your proposal.

**Comment by s-ohnishi** [comment]
After searching for the error message, I found an article about running out of memory.
Therefore, when I temporarily increased the memory to 2GB and ran it, I was able to compile without problems in about 15 minutes.

In my case, 2GB worked fine, but it may be because I haven't installed anything yet since I just started the VPS.
I'm using Debian, not Ubuntu, so maybe that's why.

I'm guessing you'll be able to install without problems if you temporarily allocate more memory.

**Comment by starjackioaz** [comment]
After perusing your most recent responses and experimenting with the **<a href="https://run3donline.com/">run 3 online</a>** game, I believe I now have a fundamental understanding of how it operates. I look forward to conducting additional tests to determine how I can make everything function.Many thanks to each of you!
